persistent/Database/Persist/Sql/Internal.hs

```diff
@@ -52,13 +63,15 @@ mkColumns allDefs t =
                            show d ++ " on " ++ show tn
         | otherwise = maxLen as
 
+    refNameFn = fromMaybe refName (backendSpecificForeignKeyName overrides)
+
     ref :: DBName
         -> ReferenceDef
         -> [Attr]
         -> Maybe (DBName, DBName) -- table name, constraint name
     ref c fe []
         | ForeignRef f _ <- fe =
-            Just (resolveTableName allDefs f, refName tn c)
+            Just (resolveTableName allDefs f, refNameFn tn c)
```

Comment:
Ah! We have been ignoring the user-provided name of the foreign key reference this entire time?? Dang. Welp. Fixing that is out-of-scope for this PR.

Reply:
Oh, hah, didn’t even notice

Reply:
maybe the constraint name override can be pushed to master now since that is a backwards compatible workaround. We only implemented it for MySQL since that is the backend we use and were running into the length issue but didn't have the resources to dedicate towards a proper solution.